subworkflows/nf-core/abundance_differential_filter/main.nf

```diff
@@ -26,9 +27,12 @@ workflow ABUNDANCE_DIFFERENTIAL_FILTER {
     ch_transcript_lengths    // [ meta_exp, transcript_lengths]
     ch_control_features      // [meta_exp, control_features]
     ch_contrasts             // [[ meta_contrast, contrast_variable, reference, target ]]
+    ch_contrasts_dream       // [meta_contrast, samplesheet, matrix]
```

Comment:
In fact, why does it have its own input as well?

Reply:
I was transferring the work from the previous PR into this one, and while this point wasn’t raised earlier during its review, I agree with your observation. I’m not aware of the reason behind this decision.

My next steps are to remove this from the subworkflow. If I find a blocker that justifies having it as its own channel, I’ll update here.

Reply:
I believe it's simply that the new structure hadn't been adopted yet when the tmp_dev branch was developed...

Reply:
I did some investigation into why `ch_contrasts` might not have been used, and I believe the issue stems from how the YAML contrasts file is handled in the `VALIDATOR` module. Specifically, the formula field is not emitted as part of the `VALIDATOR.out.contrasts` and is therefore lost. 

I propose the following to move forwards:
-  Modify `shinyngs/validatefomcomponents` R script to emit a new column, formula, in YAML contrast files. The output in `VALIDATOR.out.contrasts` will then be:
```
id	variable	reference	target	blocking	formula
treatment_mCherry_hND6_sample_number	treatment	mCherry	hND6	sample_number	~ treatment + sample_number
```
We'll then have `meta.formula` in `ch_contrasts` (and I'll reassess how to continue from there).

Reply:
Quick update: I’ve [opened a PR to shinyngs](https://github.com/pinin4fjords/shinyngs/pull/72) to address the missing formula field issue. I’ve also already started making some changes in this draft adjusting the subworkflows and preparing for integration.

Once the `shinyngs` update is merged, I’ll test it here, complete the necessary subworkflow modifications, and add a new test case. If everything works as expected and there’s agreement on the changes, I’ll move on to updating the corresponding subworkflows upstream in nf-core/modules.